File: references/java-sdk/src/main/java/blue/language/sdk/SimpleDocBuilder.java
package blue.language.sdk;

import blue.language.model.Node;

public final class SimpleDocBuilder extends DocBuilder<SimpleDocBuilder> {

    private SimpleDocBuilder() {
        super();
    }

    private SimpleDocBuilder(Node existingDocument) {
        super(existingDocument);
    }

    public static SimpleDocBuilder doc() {
        return new SimpleDocBuilder();
    }

    public static SimpleDocBuilder edit(Node existingDocument) {
        return new SimpleDocBuilder(existingDocument);
    }

    public static SimpleDocBuilder from(Node existingDocument) {
        if (existingDocument == null) {
            throw new IllegalArgumentException("existing document is required");
        }
        return new SimpleDocBuilder(existingDocument.clone());
    }
}

File: libs/dsl-sdk/src/lib/doc-builder/simple-doc-builder.ts
import type { BlueNode } from '@blue-labs/language';
import type { JsonObject } from '../core/types.js';
import { DocBuilder } from './doc-builder.js';

export class SimpleDocBuilder extends DocBuilder {
  private constructor(initial?: JsonObject) {
    super(initial);
  }

  static override doc(): SimpleDocBuilder {
    return new SimpleDocBuilder();
  }

  static override edit(
    existingDocument: JsonObject | BlueNode,
  ): SimpleDocBuilder {
    return new SimpleDocBuilder(this.documentToJson(existingDocument));
  }

  static override from(
    existingDocument: JsonObject | BlueNode,
  ): SimpleDocBuilder {
    return new SimpleDocBuilder(this.documentToJson(existingDocument));
  }
}
